src/App/Controls/PasswordStrengthProgressBar/PasswordStrengthViewModel.cs

```diff
@@ -0,0 +1,82 @@
+﻿using System.Collections.Generic;
+using Bit.Core.Abstractions;
+using Bit.Core.Utilities;
+using Xamarin.Forms;
+
+namespace Bit.App.Controls
+{
+    public class PasswordStrengthViewModel : ExtendedViewModel
+    {
+        private readonly IPasswordGenerationService _passwordGenerationService;
+        private readonly IPasswordStrengthable _passwordStrengthable;
+        private double _passwordStrength;
+        private Color _passwordColor;
+        private PasswordStrengthLevel? _passwordStrengthLevel;
+
+        public PasswordStrengthViewModel(IPasswordStrengthable passwordStrengthable)
+        {
+            _passwordGenerationService = ServiceContainer.Resolve<IPasswordGenerationService>();
+            _passwordStrengthable = passwordStrengthable;
+        }
+
+        public double PasswordStrength
+        {
+            get => _passwordStrength;
+            set => SetProperty(ref _passwordStrength, value);
+        }
+
+        public Color PasswordColor
+        {
+            get => _passwordColor;
+            set => SetProperty(ref _passwordColor, value);
+        }
+
+        public PasswordStrengthLevel? PasswordStrengthLevel
+        {
+            get => _passwordStrengthLevel;
+            set => SetProperty(ref _passwordStrengthLevel, value);
+        }
+        public List<string> GetPasswordStrengthUserInput(string email) => _passwordGenerationService.GetPasswordStrengthUserInput(email);
+
+        public Color VeryWeakColor { get; set; }
+        public Color WeakColor { get; set; }
+        public Color GoodColor { get; set; }
+        public Color StrongColor { get; set; }
```

Comment:
Remove this when implementing the change of `VeryWeak` and new bindable property